src/vmm/src/lib.rs

```diff
@@ -996,25 +952,30 @@ mod tests {
     #[test]
     fn test_create_vcpus() {
         // The scopes force the created vCPUs to unmap their kernel memory at the end.
-        {
-            let vmm_config = default_vmm_config();
-            let mut vmm = mock_vmm(vmm_config);
+        let mut vmm_config = default_vmm_config();
+        vmm_config.vcpu_config = VcpuConfig { num: 0 };
 
-            // Creating 0 vCPUs throws an error.
+        // Creating 0 vCPUs throws an error.
+        {
             assert!(matches!(
-                vmm.create_vcpus(&VcpuConfig { num: 0 }),
-                Err(Error::VcpuNumber(0))
+                Vmm::try_from(vmm_config.clone()),
+                Err(Error::Vm(vm::Error::CreateVmConfig(
+                    vm_vcpu::vcpu::Error::VcpuNumber(0)
+                )))
             ));
+        }
 
-            // Creating one works.
-            assert!(vmm.create_vcpus(&VcpuConfig { num: 1 }).is_ok());
+        // Creating one works.
+        vmm_config.vcpu_config = VcpuConfig { num: 1 };
+        {
+            assert!(Vmm::try_from(vmm_config.clone()).is_ok());
         }
 
+        // Creating 254 also works (that's the maximum number on x86 when using MP Table).
+        vmm_config.vcpu_config = VcpuConfig { num: 254 };
         {
-            let vmm_config = default_vmm_config();
-            let mut vmm = mock_vmm(vmm_config);
-            // Creating 255 also works.
-            assert!(vmm.create_vcpus(&VcpuConfig { num: u8::MAX }).is_ok());
+            Vmm::try_from(vmm_config.clone()).unwrap();
+            assert!(Vmm::try_from(vmm_config.clone()).is_ok());
```

Comment:
It was failing in the tests, and I did not know why. I think I'll just keep the `unwrap` in the test because when things don't work we have a better idea of what actually failed.

Reply:
Done. We're now just keeping the unwrap.